pkg/aws/subnets/reconciler_delete.go

```diff
@@ -0,0 +1,50 @@
+package subnets
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/aws-vpc-operator/pkg/aws"
+	"github.com/giantswarm/aws-vpc-operator/pkg/errors"
+)
+
+func (r *reconciler) ReconcileDelete(ctx context.Context, request aws.ReconcileRequest[[]aws.DeletedCloudResourceSpec]) (err error) {
```

Comment:
Why do the delete reconcilers use these generics types for the request? But the normal doesn't? Why is it a slice?

Reply:
> Why do the delete reconcilers use these generics types for the request? But the normal doesn't? 

Gradual refactoring towards all reconcilers having the same approach with generics (so there is less copy paste code).

Reply:
> Why is it a slice?

Specifying the subnets that should be deleted.

Reply:
It will all make sense in a PR or two :smile: